src/main/java/gov/cabinetoffice/gap/adminbackend/mappers/SchemeMapper.java

```diff
@@ -1,40 +1,144 @@
+    private UserServiceConfig userServiceConfig;
+    private WebClient.Builder webClientBuilder;
 
-    SchemeEntity schemeDtoToEntity(SchemeDTO schemeDto);
+    @Autowired
+    public final void setUserService(final UserService userService) {
+        this.userService = userService;
+    }
 
-    List<SchemeEntity> schemeDtoListToEntityList(List<SchemeDTO> schemeDto);
+    @Autowired
+    public final void setEncryptionService(AwsEncryptionServiceImpl encryptionService) {
+        this.encryptionService = encryptionService;
+    }
+
+    @Autowired
+    public final void setWebClientBuilder(WebClient.Builder webClientBuilder) {
+        this.webClientBuilder = webClientBuilder;
+    }
+
+    @Autowired
+    public final void setLambdaSecretConfigProperties(LambdaSecretConfigProperties lambdaSecretConfigProperties) {
+        this.lambdaSecretConfigProperties = lambdaSecretConfigProperties;
+    }
+
+    @Autowired
+    public final void setUserServiceConfig(UserServiceConfig userServiceConfig) {
+        this.userServiceConfig = userServiceConfig;
+    }
+
+    public SchemeDTO schemeEntityToDto(SchemeEntity schemeEntity) {
+        if ( schemeEntity == null ) {
+            return null;
+        }
+
+        SchemeDTO.SchemeDTOBuilder schemeDTO = SchemeDTO.builder();
+
+        schemeDTO.schemeId( schemeEntity.getId() );
+        schemeDTO.ggisReference( schemeEntity.getGgisIdentifier() );
+        schemeDTO.contactEmail( schemeEntity.getEmail() );
+        schemeDTO.funderId( schemeEntity.getFunderId() );
+        schemeDTO.name( schemeEntity.getName() );
+        if ( schemeEntity.getVersion() != null ) {
+            schemeDTO.version( String.valueOf( schemeEntity.getVersion() ) );
+        }
+        schemeDTO.createdDate( schemeEntity.getCreatedDate() );
+        schemeDTO.createdBy( schemeEntity.getCreatedBy() );
+
+        // Slightly strange logic here but if "last updated by" is null then we want to return "created by" instead.
+        // Considering the above, it doesn't make sense to potentially return mismatched updaters/update dates
+        // SO, if last updated by is not null then return this pair of values, otherwise return created by and created date
+        final boolean isLastUpdatedBySet = schemeEntity.getLastUpdatedBy() != null;
+        final int lastUpdatedBy  = isLastUpdatedBySet ? schemeEntity.getLastUpdatedBy() : schemeEntity.getCreatedBy();
+        final String lastUpdatedByEmail = userService.getGrantAdminById(lastUpdatedBy)
+                .map(admin -> {
+                    final String sub = admin.getGapUser().getUserSub();
+                    return getEmailAddressForSub(sub);
+                })
+                .orElse(EMPTY_EMAIL_VALUE); // Should literally never end up in here but would rather display a blank value than throw an error
+        final Instant lastUpdatedDate = isLastUpdatedBySet ? schemeEntity.getLastUpdated() : schemeEntity.getCreatedDate();
+
```

Comment:
Could be worth extracting the above 10 lines into a function to keep this method clean

Reply:
done